app/models/pawn.rb

```diff
@@ -1,6 +1,127 @@
 class Pawn < Piece
-  def move_valid?
-    # add logic here
+  def move_valid?(x, y)
+    return true if one_step_legal?(x, y)
+    return true if two_step_first_move_legal?(x, y)
+    return true if regular_capture_legal?(x, y)
+    return true if capture_en_passant_legal?(x, y)
+    false
+  end
+
+  # basic one-step move:
+  def one_step_legal?(x, y)
+    return false if space_occupied?(x, y)
+    return true if black_one_step_is_ok?(x, y) || white_one_step_is_ok?(x, y)
+    false
+  end
+
+  def black_one_step_is_ok?(x, y)
+    return true if is_black && (x_position == x && (y_position - y) == 1)
+    false
+  end
+
+  def white_one_step_is_ok?(x, y)
+    return true if is_black == false && (x_position == x && (y - y_position) == 1)
+    false
+  end
+
+  # two-step first move:
+  def two_step_first_move_legal?(x, y)
+    return false if space_occupied?(x, y)
+    return true if black_two_step_first_move_is_ok?(x, y) || white_two_step_first_move_is_ok?(x, y)
```

Comment:
(In the black_two_step_first_move_is_ok? method I require that the y position is 6;
In the white_two_step_first_move_is_ok? method i require that the y position is 1.)